be2-scala/src/main/scala/ch/epfl/pop/pubsub/graph/DbActor.scala

```diff
@@ -33,6 +36,15 @@ object DbActor extends AskPatternConstants {
    */
   final case class Write(channel: Channel, message: Message) extends Event
 
+  /**
+   * Request to write a message and a LaoData object in the database
+   *
+   * @param channel the channel where the message should be published
+   * @param message the message to write in the database
+   * @param obj the LaoData we write
+   */
+  final case class WriteLaoData(channel: Channel, message: Message, obj: LaoData) extends Event 
```

Comment:
My initial idea was that since every object (LaoData and the future ElectionData) will always be accompagnied by a message, to keep the database consistent and achieve atomic writing, we should therefore not allow one of the two (message or object) to be lost and the other written in the database, but both should fail at once or be written at once, therefore I write both.

Reply:
Ensuring that the changes happens atomically makes a lot of sense.

My concern is that:
- the API name "WriteLaoData" is not aligned with the parameters (channel, message)
- the API doesn't seem to restrict which messages are allowed to change LaoData
- the modification of the LaoData could be entirely unrelated to the message itself.

Effectively, a buggy user of this API could write a message (say, a chirp) and change the LaoData to remove all attendees, and nothing would stop it.

I'm not saying the following is a good idea, as a matter of fact it might be very bad practice in Scala (I wouldn't know), but imagine the following solution, in which Write is used instead of WriteLaoData. 
Pseudocode:
```
Write(channel: Channel, message: Message) {
   // check if lao data would be modified by message, through a static function
   if ( ! LaoData.IsAffectedBy(message) ) {
     save(message) // store in db, directly
   }
   else { 
     startBatch()
     laoData = retrieveLaoData(channel) // get current lao data from db
     laoData.updateWith(message) // updates LaoData as a function of message
     save(laoData) // store in db
     save(message) // store in db
     executeBatch() // finish transaction
   }
}
```

This would make the DbActor a bit "smarter" (not just a plain storage layer), but it makes sure that no amount of buggy code can mess up the database.

On a side note, you'll realize that the code above can be a source of concurrency issues: if multiple threads change the LaoData in parallel, they might end up overwriting each other. It might be worthwhile to make sure that the LaoData read-update-write cycle doesn't happen concurrently.

Reply:
This seems like a good method. Normally, only a CloseRollCall will change LaoData and it should not happen in parallel with another one, however, we should probably not assume such things, so I believe some kind of compare-and-swap instruction could be useful. However, the biggest concurrency issue I see here for now is having the entire LaoData overwritten by one thread, which with the current functionalities should not create inconsistencies, since there shouldn't be any modifications done on the existing LaoData (the attendee list is reset every time and the owner should not change)

Reply:
However, I'm still a little hesitant about adding the additional check to write, since it will complexify the method even more and take away some utility from the handlers, however having a single write method would be easier and more secure so I'll go with that.